api/v1alpha1/amaltheasession_types.go

```diff
@@ -551,6 +555,15 @@ type ReadinessProbe struct {
 	Type ReadinessProbeType `json:"type,omitempty"`
 }
 
+type Template struct {
+	Metadata TemplateMetadata `json:"metadata,omitzero"`
```

Comment:
I think I finally found what breaks with these changes:

```suggestion
	// +optional
	Metadata TemplateMetadata `json:"metadata,omitzero"`
```

Without this, the validator always expects to have a `metadata` field, which you can see fails the tests. If you add the optional, it should remove the `required` from the generated schema and that should make the CRD changes compatible with the existing custom resources.